### DISCUSSION TITLE
Error connecting to [myinstance]: The received certificate has expired

### DISCUSSION THREAD

**Discussion by acappelletti** [dataplat/dbatools#8001] [Q&A] [answered]
Hi, don't understood the meaning of this error, calling the command below:

Get-DbaDbSpace -SqlInstance myinstance  | Format-Table

Error connecting to [myinstance  ]: The received certificate has expired

I got tens instances and only in this one I have problem with dbatools.
All it's fine with SSMS connection

Thanks Alen

**Comment by niphlod** [comment]
seems like there's an expired cert serverside. did you try connecting with SSMS checking if the cert is valid (Encrypt connection ticked on advanced tab, trust server certificate unticked ) ?

**Reply by acappelletti** [reply]
Yes I tried from SSMS with no problem...

**Reply by niphlod** [reply]
so you are using encryption in SSMS ?

**Reply by acappelletti** [reply]
![image](https://user-images.githubusercontent.com/43847440/144402021-86bc76d2-40d8-4c6a-ab71-001221e9f636.png)

**Reply by acappelletti** [reply]
With or without these ticks my connection is right from SSMS

**Reply by niphlod** [reply]
what if you try to do a Connect-DbaInstance myinstance ? try fiddling with -EncryptConnection and -TrustServerCertificate to see if the error goes away

**Reply by acappelletti** [reply]
Yes, in this way it worked .. but I would be curious to understand why. Then I solved it like this
Connect-DbaInstance 'MyInstance'  -EncryptConnection  -TrustServerCertificate | Get-DbaDbSpace | Format-Table

 Thanks Alen...

**Reply by mlakson** [reply]
I have a very similar problem. I have two sql instances on windows cluster, both with ssl encryption. I can connect to named instance on the same server without any problems. Default instance returns error. All it's fine with SSMS connection.

Test-DbaDiskSpeed -SqlInstance 'Instance.contoso.com'
WARNING: [15:04:42][Test-DbaDiskSpeed] Failure | Error connecting to [Instance.contoso.com]: The target principal name is
 incorrect

Workaround with Connect-DbaInstance works but I still can't make a backup 

Connect-DbaInstance 'Instance.contoso.com' -EncryptConnection -TrustServerCertificate | Backup-DbaDatabase -Database TestDB -BackupDirectory D:\SQLBackup
WARNING: [15:16:20][Backup-DbaDatabase] No databases match the request for backups

Thanks Igor

**Comment by potatoqualitee** [comment, marked as answer]
This relates to the pinned Discussion -- Microsoft changed the way things work to be more secure, which I appreciate. But this can be overridden in our configs without having to specify the trust over and over. Check out the "New SMO Library and Certificate Trust" post for more information.

https://github.com/dataplat/dbatools/discussions/7680

**Reply by mlakson** [reply]
thanks, now everything is working
